cpp/src/join/join_compute_api.h

```diff
@@ -382,13 +409,12 @@ gdf_error compute_hash_join(
   gdf_table<size_type> const & probe_table{left_table};
   const size_type probe_table_num_rows{probe_table.get_column_length()};
 
-  // Calculate size of hash map based on the desired occupancy
-  size_type hash_table_size{(build_table_num_rows * 100) / DEFAULT_HASH_TABLE_OCCUPANCY};
+  // Hash table size must be at least 1 in order to have a valid allocation.
+  // Even if the hash table will be empty, it still must be allocated for the
+  // probing phase in the event of an outer join
+  size_t const hash_table_size =
+      std::max(compute_hash_table_size(build_table_num_rows), size_t{1});
```

Comment:
Presumably you won't get this far if the build table size is zero rows, so couldn't you move this (the `max(hash_table_size, 1)`) inside `compute_hash_table_size()`?

Reply:
If `build_table_num_rows==0` you'll still get this far. 

It's perfectly valid for `compute_hash_table_size` to return 0. I prefer to keep it that way to keep the function more generic and reusable. 

The fact that I need a hash table size of at least `1` is special logic for this use case. Otherwise it would be impossible to tell the difference between `compute_hash_table_size(0)` and `compute_hash_table_size(2, 50)`.